### PR TITLE
Fix: Lua editor can open the same file repeatedly

### PR DESCRIPTION
**Pull request by tjmgd**
Lua editor can open the same file repeatedly through different opening methods

Signed-off-by: T.J. McGrath-Daly <tj.mcgrath.daly@huawei.com>

### CODE REVIEW

File: Code/Tools/Standalone/Source/LUA/LUAEditorMainWindow.cpp [hunk 1]
```diff
@@ -345,8 +345,17 @@ namespace LUAEditor
...
-                const AZStd::string filePath = selectedAsset->GetFullPath();
-                EBUS_EVENT(Context_DocumentManagement::Bus, OnLoadDocument, filePath, true);
+
+                AZStd::string current = selectedAsset->GetFullPath();
+                AZStd::replace(current.begin(), current.end(), '\\', '/');
+
+                const AZStd::string filePath = current;
+
+                auto entry_type = selectedAsset->GetEntryType();
+                if (entry_type == AzToolsFramework::AssetBrowser::AssetBrowserEntry::AssetEntryType::Source)
```
Comment (lumberyard-employee-dm) [inline Code/Tools/Standalone/Source/LUA/LUAEditorMainWindow.cpp:355]: standards: The variable naming standard should be [lowerCamelCase](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names)

**Review by lsemp3d — CHANGES_REQUESTED** [review]
Correct the variable name convention per lumberyard-employee-dm's review

File: Code/Tools/Standalone/Source/LUA/LUAEditorMainWindow.cpp [hunk 2]
```diff
@@ -345,8 +345,17 @@ namespace LUAEditor
...
+                {
+                    EBUS_EVENT(Context_DocumentManagement::Bus, OnLoadDocument, filePath, true);
```
Comment (cgalvan) [inline Code/Tools/Standalone/Source/LUA/LUAEditorMainWindow.cpp:357]: While we're moving this code, we should also update to not use the old macro style ebus call. The updated one would look like:

```
Context_DocumentManagement::Bus::Broadcast(&Context_DocumentManagement::OnLoadDocument, filePath, true);
```

**Review by cgalvan — CHANGES_REQUESTED** [review]

**Comment by hultonha** [comment]
Any update on this @tjmgd? Looks like some feedback was requested but there hasn't been an update. If you ping me when the updates have been made I can kick off a build. Thanks!

**Comment by tjmgd** [comment]
> Any update on this @tjmgd? Looks like some feedback was requested but there hasn't been an update. If you ping me when the updates have been made I can kick off a build. Thanks!

I am closing this PR and opening a new one once the changes are made. Ran into git problems!

**Comment by hultonha** [comment]
> > Any update on this @tjmgd? Looks like some feedback was requested but there hasn't been an update. If you ping me when the updates have been made I can kick off a build. Thanks!
> 
> I am closing this PR and opening a new one once the changes are made. Ran into git problems!

No problem, thanks for the update @tjmgd, I'd go ahead and add the original reviewers and myself when it's posted. Thanks!

**Pull request closed without merging**